card/mod/05_standard/set/all/links.rb

```diff
@@ -31,7 +31,11 @@ def web_link href, opts={}
   # link to a specific card
   def card_link name, opts={}
     text = (opts.delete(:text) || name).to_name.to_show @context_names
-    name = Card[name].cardname if Symbol===name
+    name = case name
+           when Symbol then Card[name].cardname
+           when Card   then name.cardname
+           else             name
+           end
```

Comment:
Scratch the comment about should work if it doesn't exist.  My proposal wouldn't make that work.  There can't be a codename without a card.  Still good to avoid any Card stuff for that case.